From b9d40acc4e06ddd2abfd067dd3bed2f40ebe9bd3 Mon Sep 17 00:00:00 2001
From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Wed, 9 Dec 2015 13:41:29 +0300
Subject: mlxsw: core: remove an unneeded condition
Reply-To: joseph.salisbury@microsoft.com

We already know "err" is zero so there is no need to check.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Acked-by: Jiri Pirko <jiri@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 515123e286b686d32256342318f20ff932ebac1b)
Signed-off-by: Joseph Salisbury <joseph.salisbury@microsoft.com>
---
 drivers/net/ethernet/mellanox/mlxsw/core_hwmon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/core_hwmon.c b/drivers/net/ethernet/mellanox/mlxsw/core_hwmon.c
index 4dad146b41ae..913106d37bb0 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/core_hwmon.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/core_hwmon.c
@@ -169,7 +169,7 @@ static ssize_t mlxsw_hwmon_pwm_store(struct device *dev,
 		dev_err(mlxsw_hwmon->bus_info->dev, "Failed to write PWM\n");
 		return err;
 	}
-	return err ? err : len;
+	return len;
 }
 
 enum mlxsw_hwmon_attr_type {
-- 
2.17.1

